_dev/sources/miscBuilder.ts

```diff
@@ -71,19 +114,32 @@ export class MiscBuilder {
                 userMapProviderId = currentUser.defaultMapEngine,
                 defaultMapProviderId = systemSettings.mapProvider,
                 mapProviderId = this.getMapProviderType(userMapProviderId) === "custom" ? userMapProviderId : defaultMapProviderId;
+            this.purgeSettings = systemSettings.purgeSettings;
+            this.emailSenderFrom = systemSettings.emailSenderFrom;
+            this.customerClassification = systemSettings.customerClassification;
             this.currentUser = currentUser;
             this.customMapProviders = entityToDictionary(systemSettings.customWebMapProviderList);
             this.isUnsignedAddinsAllowed = systemSettings.allowUnsignedAddIn;
             this.addins = this.getAllowedAddins(systemSettings.customerPages);
-            return {
+            this.isMarketplacePurchasesAllowed = systemSettings.allowMarketplacePurchases;
+            this.isResellerAutoLoginAllowed = systemSettings.allowResellerAutoLogin;
+            this.isThirdPartyMarketplaceAppsAllowed = systemSettings.allowThirdPartyMarketplaceApps;
+            const output = {
                 mapProvider: {
                     value: mapProviderId,
                     type: this.getMapProviderType(mapProviderId)
                 },
                 currentUser: this.currentUser,
                 isUnsignedAddinsAllowed: this.isUnsignedAddinsAllowed,
-                addins: this.addins
+                addins: this.addins,
+                ...(includeSysSettings && { purgeSettings: this.purgeSettings }),
```

Comment:
Type checking doesn't work for such construction.
Just add condition below instead this:
```
if (includeSysSettings) {
    output.emailSenderFrom = this.emailSenderFrom,
    .........
}
```

Reply:
It does work - I have tested it and it works fine!

Reply:
I know that it works. But type checker doesn't check types when you use spread operator.

Reply:
who's checking types? I am setting a value based on a condition...not setting a type???

Reply:
(condition) && {propert=value}

Reply:
Do you mean this:
            let output = {
                mapProvider: {
                    value: mapProviderId,
                    type: this.getMapProviderType(mapProviderId)
                },
                currentUser: this.currentUser,
                isUnsignedAddinsAllowed: this.isUnsignedAddinsAllowed,
                addins: this.addins,
                // purgeSettings: this.purgeSettings,
                // emailSenderFrom: this.emailSenderFrom,
                // customerClassification: this.customerClassification,
                // isMarketplacePurchasesAllowed: this.isThirdPartyMarketplaceAppsAllowed,
                // isResellerAutoLoginAllowed: this.isResellerAutoLoginAllowed,
                // isThirdPartyMarketplaceAppsAllowed: this.isThirdPartyMarketplaceAppsAllowed
            };
            if (includeSysSettings) {
                output.purgeSettings = this.purgeSettings;
                output.emailSenderFrom = this.emailSenderFrom;
                output.customerClassification = this.customerClassification;
                output.isMarketplacePurchasesAllowed = this.isMarketplacePurchasesAllowed;
                output.isResellerAutoLoginAllowed = this.isResellerAutoLoginAllowed;
                output.isThirdPartyMarketplaceAppsAllowed = this.isThirdPartyMarketplaceAppsAllowed;
            }

Reply:
Typescript checks types. You can make a typo in property name (e.g. `...(includeSysSettings && { **purge**: this.purgeSettings }),`). Resulted object will be inconsistent with IMiscData interface but ts compiler will not produce any errors during compilation of your code.

Reply:
done